library/src/main/kotlin/com/livefront/gsonkotlinadapter/KoltinReflectiveTypeAdapterFactory.kt

```diff
@@ -38,17 +41,30 @@ class KotlinReflectiveTypeAdapterFactory private constructor() : TypeAdapterFact
         require(!kotlinRawType.isSealed) { "Cannot serialize sealed class ${rawType.name}" }
         require(!kotlinRawType.isInner) { "Cannot serialize inner class ${rawType.name}" }
         kotlinRawType.primaryConstructor ?: return null
-        return Adapter(this, gson, type, kotlinRawType)
+        return Adapter(this, gson, type, kotlinRawType, enableDefaultPrimitiveValues)
     }
 
     internal class Adapter<T : Any>(
         factory: TypeAdapterFactory,
         gson: Gson,
         type: TypeToken<T>,
-        kClass: KClass<T>
+        kClass: KClass<T>,
+        private val enableDefaultPrimitiveValues: Boolean
     ) : TypeAdapter<T>() {
         private val primaryConstructor: KFunction<T> = kClass.primaryConstructor!!
         private val declaringClass: Class<T> = primaryConstructor.javaConstructor?.declaringClass!!
+        private val constructorParameterDefaultsMap: Map<KParameter, Any?> = primaryConstructor
```

Comment:
I'm wondering if this name is a little misleading because its not actually finding all the defaults, since this is only for primitives. Should you just make the name more specific? `constructorPrimitiveDefaultsMaps` or something?

Reply:
So one thing that this map also does is add in a default value of `null` for any nullable, non-optional properties that aren't in the JSON, even when default primitive values is disabled, so I think the name as-is makes sense. Maybe those two concerns should be separated though?

Reply:
It does have all the defaults for non-primitives too, they just happen to be null.

The only parameters without defaults in this map are the `optional` ones since we need to make sure that the kotlin defaults get used.

Reply:
Hmmm, maybe `defaultPrimitiveValue` should become `defaultValue`?